apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java

```diff
@@ -313,61 +315,29 @@ public ApikeyController(ApikeyRepo apikeyRepo) {
      * Validates a given Apikey. Sets last access date and activation date (if not set, ie. first access) with the
      * current date and +1 increments the usage count of this Apikey.
      *
-     * @param   id     the apikey to validate
-     * @param   api    API for which validate this apikey
-     * @param   method method (read, write) for which validate this apikey
+     * @param   httpServletRequest     request
      *
      * @return  HTTP 204 upon successful validation
-     *          HTTP 400 if a mandatory parameter is missing
-     *          HTTP 401 in case of an invalid request
-     *          HTTP 403 if the request is unauthorised
-     *          HTTP 404 when the requested Apikey is not found in the database
+     *          HTTP 400 bad request when header does not contain api key
+     *          HTTP 401 in case of an unregistered api key
      *          HTTP 410 when the requested Apikey is deprecated (i.e. has a past deprecationdate)
-     *          HTTP 429 if the assigned usagelimit has been reached
-     *          Addionally, the following fields are (optionally) available in the response header:
-     *          - "X-RateLimit-Remaining" access usage number since the previous reset
-     *          - "X-RateLimit-Reset"     the number of seconds until the access usage count is reset
-     *          - "Apikey-not-found"      containing the string "apikey-not-found" is added when the Apikey
-     *                                    is not found, to help telling this HTTP 404 apart from one returned
-     *                                    by the webserver for other reasons
      */
-    @RequestMapping(path = "/{id}/validate", method = RequestMethod.POST)
-    public ResponseEntity<Apikey> validate(@PathVariable("id") String id,
-                                           @RequestParam(value = "api", required = false) String api,
-                                           @RequestParam(value = "method", required = false) String method) {
-
-        LOG.debug("validate apikey: {}", id);
-        ApiName     apiName; //TODO usage not implemented yet
-        HttpHeaders headers  = new HttpHeaders();
-        DateTime    nowDtUtc = new DateTime(DateTimeZone.UTC);
-        Date        now      = nowDtUtc.toDate();
-
-        if (StringUtils.isEmpty(api)) {
-            LOG.debug("no value for parameter 'api' supplied");
+    @PostMapping(path = "/validate")
+    public ResponseEntity<Apikey> validate(HttpServletRequest httpServletRequest) {
+        // When no apikey was supplied return 400
+        String id = getApikey(httpServletRequest);
+        if (null == id) {
+            LOG.debug(APIKEYMISSING);
```

Comment:
the bad request response should also indicate the reason. (e.g. Authorization header required)